libarchive/archive_read_support_filter_gzip.c

```diff
@@ -209,7 +209,7 @@ peek_at_header(struct archive_read_filter *filter, int *pbits,
 	}
 
 	/* Optional header CRC */
-	if ((header_flags & 2)) {
+	if (header_flags & 2) {
```

Comment:
The extra parentheses are needed here.   That's used by several C compilers to indicate that you're doing a boolean check.